src/main/java/edu/kit/ipd/crowdcontrol/objectservice/feedback/FeedbackSender.java

```diff
@@ -0,0 +1,132 @@
+        Optional<ExperimentRecord> exp = expOps.getExperiment(expId);
+        if (exp.isPresent()) {
+            map.put("experimentName", exp.get().getTitle());
+        } else {
+            LOGGER.error("Error, experiment cannot be found.");
+            throw new FeedbackException("Experiment cannot be found.");
+        }
+
+        //iterate over answers and send them and the feedback to the workers
+        for (AnswerRecord answer : answers) {
+            //List all ratings to an answer in the message.
+            List<RatingRecord> ratings = answerOps.getRatingsOfAnswer(answer);
+
+            StringBuilder ratingMessage = new StringBuilder();
+
+            for (RatingRecord rating : ratings) {
+                Map<String, String> ratingMap = new HashMap<>();
+                ratingMap.put("feedback", rating.getFeedback());
+                ratingMap.put("quality", rating.getQuality().toString());
+                ratingMap.put("rating", rating.getRating().toString());
+
+                ratingMessage.append(Template.apply(feedbackRating, ratingMap));
+            }
+
+            //Replace placeholders with answer and the ratings
+            Map<String, String> answerMap = new HashMap<>();
+            answerMap.put("answer", answer.getAnswer());
+            answerMap.put("ratings", ratingMessage.toString());
+
+            answerMessage.append(Template.apply(feedbackAnswer, answerMap));
+        }
+
+        //Send feedback to the last worker
+        String message = "";
+        if (!answers.isEmpty()) {
+            map.put("answers", answerMessage.toString());
+            message = Template.apply(feedbackMessage, map);
+        }
+        LOGGER.trace("Completed creating feedback message to worker " + workerId + ".");
+        return message;
+    }
+
+    static String loadMessage(String path) throws FeedbackException {
+        StringBuilder content = new StringBuilder();
+
+        try {
+            FileReader file = new FileReader(path);
+            BufferedReader reader = new BufferedReader(file);
+            String messageLine;
+            while ((messageLine = reader.readLine()) != null) {
+                content.append(messageLine);
+                content.append(System.getProperty("line.separator"));
+            }
+        } catch (FileNotFoundException e) {
+            LOGGER.error("Sending of feedback failed, because the message sources cannot be found.");
+            throw new FeedbackException("The file at \"" + path + "\" couldn't be found. Please secure, that there is a file.");
+        } catch (IOException e) {
+            LOGGER.error("Sending of feedback failed, because the message sources are invalid.");
+            throw new FeedbackException("The file at \"" + path + "\" couldn't be read. Please secure, that the file isn't corrupt");
+        }
+        return content.toString();
+    }
+
+    private void sendFeedback(int workerID, String workerMessage) throws FeedbackException {
```

Comment:
So maybe try something like 
```
        WorkerRecord wr = workerOps.getWorker(workerID)
                .orElseThrow(() -> new IllegalArgumentException("Worker was not found"));

        if (wr.getEmail() == null || wr.getEmail().isEmpty())
            throw new IllegalArgumentException("Worker doesnt have a emailadress");

        try {
            handler.sendMail(wr.getEmail(),"Feedback to you work", workerMessage);
        } catch (MessagingException | UnsupportedEncodingException e) {
            e.printStackTrace();
            //TODO HANDLE THIS
        }
```
Maybe you need to think about the typ of the exception, if they are correct.